src/Service/UserBookingCacheService.php

```diff
@@ -0,0 +1,246 @@
+            switch ($field) {
+                case 'title':
+                    $entity->setTitle($value);
+                    break;
+                case 'uid':
+                    break;
+                case 'start':
+                    $entity->setStart($value);
+
+                    break;
+                case 'end':
+                    $entity->setEnd($value);
+                    break;
+                case 'status':
+                    $entity->setStatus($value);
+                    break;
+                case 'resource':
+                    $entity->setResource($value);
+                    break;
+            }
+        }
+
+        $this->entityManager->flush();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteCacheEntry(string $exchangeId): void
+    {
+        $entity = $this->entityManager->getRepository(UserBookingCacheEntry::class)
+            ->findOneBy(['exchangeId' => $exchangeId]);
+        if ($entity) {
+            $this->entityManager->remove($entity);
+        }
+    }
+
+    /**
+     * Set values for cache entity.
+     *
+     * @param $entity
+     * @param \App\Entity\Main\UserBooking $userBooking
+     *
+     * @return \App\Entity\Main\UserBookingCacheEntry
+     */
+    private function setCacheEntityValues($entity, UserBooking $userBooking): UserBookingCacheEntry
+    {
+        if ($userBooking->resourceMail) {
+            $entity->setTitle($userBooking->subject);
+            $entity->setExchangeId($userBooking->id);
+            $entity->setUid($this->retrieveUidFromBody($userBooking->body) ?? '');
+            $entity->setStart($userBooking->start);
+            $entity->setEnd($userBooking->end);
+            $entity->setStatus($userBooking->status);
+            $entity->setResource($userBooking->resourceMail);
+        }
+
+        return $entity;
+    }
+
+    /**
+     * Get uid from mail body.
+     *
+     * @param string $body
+     *
+     * @return string|null
+     */
+    private function retrieveUidFromBody(string $body): ?string
+    {
+        $doc = new \DOMDocument();
+        $doc->loadHTML($body);
+        $uidDomElement = $doc->getElementById('userId');
+
+        return $this->extractRealUid($uidDomElement?->textContent);
+    }
+
+    /**
+     * Clear UserBookingCacheEntry table.
+     *
+     * @return void
+     */
+    private function clearUserBookingCache(): void
+    {
+        $repository = $this->entityManager->getRepository(UserBookingCacheEntry::class);
+        $entities = $repository->findAll();
+
+        foreach ($entities as $entity) {
+            $this->entityManager->remove($entity);
+        }
+
+        $this->entityManager->flush();
+    }
+
+    /**
+     * Remove UID from front and back of id.
+     *
+     * @param $documentBodyUid
+     *
+     * @return string
+     */
+    private function extractRealUid($documentBodyUid): string
+    {
+        $documentBodyUid = ltrim($documentBodyUid, 'UID-');
```

Comment:
Would be prettier to handle with a regexp

Reply:
CHanged, and fixed bug while doing so...